pkg/runtime/runtime.go

```diff
@@ -497,11 +504,29 @@ func (a *DaprRuntime) beginPubSub(name string, ps pubsub.PubSub) error {
 				}
 			}
 
+			if pubsub.HasExpired(cloudEvent) {
+				log.Warnf("dropping expired pub/sub event %v as of %v", cloudEvent[pubsub.IDField], cloudEvent[pubsub.ExpirationField])
+
+				return nil
+			}
+
+			route := a.topicRoutes[msg.Metadata[pubsubName]].routes[msg.Topic]
```

Comment:
@artursouza this line seems to be the cause for this bug: #3920

Is this intentional? It seems a bit off as the variable `route` is shadowed here.
With this line removed, wildcard subscriptions work again for me.

Reply:
I don't know if this line should be removed but I agree that it is odd to have route shadowed here. I think there is an opportunity for 2 contributions here:

1. Create a PR with a new E2E test that reproduces the bug you are facing.
2. Update that PR by removing/updating this line and see if that fixes the issue above without breaking other E2E tests.
3. If 2 does not work, we need to iterate to find the actual root cause.

That will give us a better idea where the problem is and raise the bar for Dapr. This fix might also trigger a hotfix depending on the severity of it.

Thanks a lot for bringing this up.

PS: Do we have an issue tracking this? If not, please, create one.

Reply:
yes removing this line is probably not the right solution, as this is inside a `func()` passed to `Subscribe` and I think this could lead to a potential data race.

#3920 is about just this issue but has no content. What do you propose, reopen and update #3920 or creating a new issue?

Reply:
I reopened #3920

Reply:
I will try to extend the e2e tests, just needed some time to understand the e2e test structure

Reply:
i created a PR for the e2e tests.https://github.com/dapr/dapr/pull/4220
and a follow up that fixes the issue in the runtime: https://github.com/thielepaul/dapr/pull/1